embedded/sql/engine.go

```diff
@@ -957,7 +957,7 @@ func DecodeValue(b []byte, colType SQLValueType) (TypedValue, int, error) {
 	vlen := int(binary.BigEndian.Uint32(b[:]))
 	voff := EncLenLen
 
-	if len(b) < vlen {
+	if vlen < 0 || len(b) < voff+vlen {
```

Comment:
vlen should be always positive, isnt?

Reply:
On 32-bit architectures (IIRC raspberry pi is one), the decoded uint32 may actually convert to negative integer when casted to int.